{
  "CVE Code": "CVE-2019-6446",
  "Published": "2019-01-16T05:00:00.000Z",
  "Updated": "2025-07-21T22:10:51.613Z",
  "Description": "An issue was discovered in NumPy before 1.16.3. It uses the pickle Python module unsafely, which allows remote attackers to execute arbitrary code via a crafted serialized object, as demonstrated by a numpy.load call. NOTE: third parties dispute this issue because it is a behavior that might have legitimate applications in (for example) loading serialized Python object arrays from trusted and authenticated sources.",
  "ProblemTypes": [
    "n/a"
  ],
  "Affected": [
    {
      "vendor": "n/a",
      "product": "n/a",
      "versions": [
        "n/a"
      ]
    }
  ],
  "Reference": [
    "https://bugzilla.suse.com/show_bug.cgi?id=1122208",
    "http://www.securityfocus.com/bid/106670",
    "https://github.com/numpy/numpy/issues/12759",
    "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/7ZZAYIQNUUYXGMKHSPEEXS4TRYFOUYE4/",
    "http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00091.html",
    "http://lists.opensuse.org/opensuse-security-announce/2019-09/msg00092.html",
    "http://lists.opensuse.org/opensuse-security-announce/2019-10/msg00015.html",
    "https://access.redhat.com/errata/RHSA-2019:3335",
    "https://access.redhat.com/errata/RHSA-2019:3704",
    "https://github.com/numpy/numpy/pull/13359",
    "https://github.com/numpy/numpy/pull/12889",
    "https://github.com/numpy/numpy/commit/89b688732b37616c9d26623f81aaee1703c30ffb"
  ]
}